### ISSUE TITLE
Bit tag fails after first version is published

### ISSUE THREAD

**Issue by lhassis** [teambit/bit#6049]
### Description
Hi everyone! We have a library built in Bit that has several react components. We've been able to create the first version of these components, tag them and publish into a private registry.

Then, after making small changes into a component, we try to tag it and update its version, but we get the following error to several components:

```
SyntaxError: C:\Users\lucas.assis\AppData\Local\Bit\capsules\03e8bb545c9cf2bb32331e6548a7bbb106166dcf\plasma_ui_themes_theme-provider@3.0.0\index.ts: Support for the experimental syntax 'flow' isn't currently enabled (2:8):

  1 | import ThemeProvider from './theme-provider';
> 2 | export type { Theme } from './theme-provider';
    |        ^
  3 | export type { ThemeProviderProps, ModeType, SizeType } from './theme-provider';
  4 | export type { ThemeProviderModel } from './theme-provider.model';
  5 | export { useThemeProviderModel } from './theme-provider.model';

Add @babel/preset-flow (https://github.com/babel/babel/tree/main/packages/babel-preset-flow) to the 'presets' section of your Babel config to enable transformation.
If you want to leave it as-is, add @babel/plugin-syntax-flow (https://github.com/babel/babel/tree/main/packages/babel-plugin-syntax-flow) to the 'plugins' section to enable parsing.
    at instantiate (C:\Users\lucas.assis\AppData\Local\.bvm\versions\0.0.749\bit-0.0.749\node_modules\@babel\parser\src\parse-error\credentials.js:61:22)
    at toParseError (C:\Users\lucas.assis\AppData\Local\.bvm\versions\0.0.749\bit-0.0.749\node_modules\@babel\parser\src\parse-error.js:58:12)
    at Object.raise (C:\Users\lucas.assis\AppData\Local\.bvm\versions\0.0.749\bit-0.0.749\node_modules\@babel\parser\src\tokenizer\index.js:1736:19)
    at Object.expectOnePlugin (C:\Users\lucas.assis\AppData\Local\.bvm\versions\0.0.749\bit-0.0.749\node_modules\@babel\parser\src\tokenizer\index.js:1800:18)
    at Object.isExportDefaultSpecifier (C:\Users\lucas.assis\AppData\Local\.bvm\versions\0.0.749\bit-0.0.749\node_modules\@babel\parser\src\parser\statement.js:2149:16)
    at Object.maybeParseExportDefaultSpecifier (C:\Users\lucas.assis\AppData\Local\.bvm\versions\0.0.749\bit-0.0.749\node_modules\@babel\parser\src\parser\statement.js:2011:14)
    at Object.parseExport (C:\Users\lucas.assis\AppData\Local\.bvm\versions\0.0.749\bit-0.0.749\node_modules\@babel\parser\src\parser\statement.js:1956:29)
    at Object.parseStatementContent (C:\Users\lucas.assis\AppData\Local\.bvm\versions\0.0.749\bit-0.0.749\node_modules\@babel\parser\src\parser\statement.js:431:25)
moduleIdentifier: C:\Users\lucas.assis\AppData\Local\.bvm\versions\0.0.749\bit-0.0.749\node_modules\babel-loader\lib\index.js??ruleSet[1].rules[1].oneOf[1]!C:\Users\lucas.assis\AppData\Local\Bit\capsules\03e8bb545c9cf2bb32331e6548a7bbb106166dcf\plasma_ui_themes_theme-provider@3.0.0\index.ts
```

### Specifications
* Bit version: 0.0.749
* Node version: 16.10.0
* npm version: 7.24.0
* Platform: Win 11
* Bit compiler (include version):
* Bit tester (include version):

### Context and additional information
All the components in error are developed under a custom react environment. We've created this environment to override the standard tsconfig, in order to accept .json files into the build process. Besides that, all the environment configurations are not changed.

If we change anything into the environment, so that all the components are updated, the components are tagged without errors.

**Comment by GiladShoham** [comment]
It seems that you import an internal file of another component from one of your components.

**Comment by KutnerUri** [comment]
this error is happening when you try to import an untranspiled file (ie regular .ts source file)

ie you are trying to access:
```
plasma_ui_themes_theme-provider@3.0.0\index.ts <-- source file
plasma_ui_themes_theme-provider@3.0.0\dist\index.js <-- dist file
```

This probably means that
(1) the component hasn't been compiled - you should be able to see it when running `bit status`
or
(2) the package.json is somehow incorrect (with a different "main" file, etc), maybe because of a "teambit.pkg/pkg" config in your workspace.jsonc.

you can troubleshoot both by looking into the component dists, at:
```
node_modules/@plasma/ui/themes.theme-provider
// or the relevant path in your project
```

**Comment by lhassis** [comment]
Thanks for the answers @GiladShoham and @KutnerUri .
But, unfortunately, none of the suggestions were my case. I've carefully checked all the imports from other components of the workspace and there are only two types of imports from other components:
importing the default
`import ComponentX from '@scope/component'`
or importing types exported from other component
`import {ComponentType} from '@scope/component'`

My "teambit.pkg/pkg" config only defines the registry
`      "packageJson": {
          "name": "@{scope}/{name}",
          "private": false,
          "publishConfig": {
            "scope": "@plasma",
            "registry": "http://localhost:8081/repository/npm-hosted/"
          }
      }`

And I've also compiled before tagging it in both scenarios: the first version (that passes) and the patch update after changing the component. Running `bit status` showed components ok and the changes, as expected.

**Comment by KutnerUri** [comment]
again, the problem is 100% because bit is accessing the source file in the capsule, instead of the missing dist file.
I amm not sure why it is missing.

can you upload the `workspace.jsonc` of the workspace, and also the env main runtime file? (`xxx.main.runtime.ts`/x)

**Comment by lhassis** [comment]
> again, the problem is 100% because bit is accessing the source file in the capsule, instead of the missing dist file. I amm not sure why it is missing.
> 
> can you upload the `workspace.jsonc` of the workspace, and also the env main runtime file? (`xxx.main.runtime.ts`/x)

Sure!

workspace.jsonc
`{
  "$schema": "https://static.bit.dev/teambit/schemas/schema.json",
  "teambit.workspace/workspace": {
    "name": "plasma-ui",
    "icon": "https://static.bit.dev/bit-logo.svg",
    "defaultDirectory": "{scope}/{name}",
    "defaultScope": "plasma"
  },
  "teambit.dependencies/dependency-resolver": {
    "packageManager": "teambit.dependencies/pnpm",
    "policy": {
      "dependencies": {
        "@fontsource/roboto": "4.5.7",
        "@sms/eslint-config-ts-recommended": "1.0.1",
        "@types/lodash": "4.14.182",
        "@types/node": "17.0.33",
        "@types/react": "17.0.5",
        "@types/react-dom": "17.0.5",
        "@types/react-grid-layout": "1.3.2",
        "@types/react-splitter-layout": "3.0.2",
        "@types/react-table": "7.7.12",
        "@types/react-virtualized": "9.21.21",
        "@types/react-window": "^1.8.5",
        "@types/styled-components": "5.1.25",
        "@typescript-eslint/eslint-plugin": "~4.10.0",
        "@typescript-eslint/parser": "~5.16.0",
        "color": "4.2.3",
        "echarts": "~5.3.3",
        "eslint": "~7.15.0",
        "eslint-import-resolver-node": "0.3.6",
        "eslint-plugin-import": "2.22.1",
        "eslint-plugin-import-helpers": "~1.1.0",
        "eslint-plugin-jest": "24.4.0",
        "eslint-plugin-jsx-a11y": "6.4.1",
        "eslint-plugin-mdx": "1.15.0",
        "eslint-plugin-prettier": "~3.3.0",
        "eslint-plugin-react": "7.25.1",
        "eslint-plugin-react-hooks": "~4.2.0",
        "lodash": "4.17.21",
        "mobx": "6.5.0",
        "mobx-react-lite": "3.4.0",
        "moment": "2.29.3",
        "prettier": "2.6.2",
        "react-dnd": "~15.1.1",
        "react-dnd-html5-backend": "~15.1.2",
        "react-grid-layout": "1.3.4",
        "react-hook-form": "7.31.1",
        "react-perfect-scrollbar": "1.5.8",
        "react-resizable": "3.0.4",
        "react-router-dom": "6.3.0",
        "react-split": "2.0.14",
        "react-table": "7.8.0",
        "react-virtualized": "9.22.3",
        "react-window": "^1.8.7",
        "resize-observer-polyfill": "~1.5.1",
        "string-similarity": "4.0.4",
        "styled-components": "5.3.5",
        "typescript": "~4.6.3",
        "vis-data": "7.1.4",
        "vis-timeline": "7.5.1"
      },
      "peerDependencies": {
        "@emotion/react": "11.9.0",
        "@emotion/styled": "11.8.1",
        "@mui/icons-material": "5.6.2",
        "@mui/material": "5.7.0",
        "@testing-library/react": "12.0.0",
        "antd": "4.20.5",
        "i18next": "21.6.16",
        "react": "17.0.2",
        "react-dom": "17.0.2",
        "react-i18next": "11.16.7",
        "react-is": "18.0.0",
        "styled-components": "5.3.5"
      }
    }
  },
  "teambit.workspace/variants": {
    "{ui/application/**}, {ui/display/**}, {ui/feedback/**}, {ui/general/**}, {ui/input/**}, {ui/layout/**}, {ui/navigation/**}, {ui/themes/**}": {
      "plasma/ui/envs/react": {},
      "teambit.pkg/pkg": {
        "packageManagerPublishArgs": [
          "--access public"
        ],
        "packageJson": {
          "name": "@{scope}/{name}",
          "private": false,
          "publishConfig": {
            "scope": "@plasma",
            "registry": "http://localhost:8081/repository/npm-hosted/"
          }
        }
      }
    },
    "{ui/utils/**}": {
      "teambit.harmony/node": {},
      "teambit.pkg/pkg": {
        "packageManagerPublishArgs": [
          "--access public"
        ],
        "packageJson": {
          "name": "@{scope}/{name}",
          "private": false,
          "publishConfig": {
            "scope": "@plasma",
            "registry": "http://localhost:8081/repository/npm-hosted/"
          }
        }
      }
    },
    "{ui/envs/**}, {ui/factory/**}": {
      "teambit.harmony/aspect": {}
    }
  },
  "teambit.generator/generator": {
    "aspects": [
      "plasma/ui/factory/react-component"
    ]
  }
}`

plasma-react.maind.runtime.ts
`import { MainRuntime } from '@teambit/cli';
import { EnvsAspect, EnvsMain } from '@teambit/envs';
import { ReactAspect, ReactMain } from '@teambit/react';
import { TsConfigTransformer, TypescriptConfigMutator } from '@teambit/typescript';

import { PlasmaReactAspect } from './plasma-react.aspect';

import 'antd/dist/antd.css';

// import { previewConfigTransformer, devServerConfigTransformer } from './webpack/webpack-transformers';

/**
 * Uncomment to include config files for overrides of Typescript or Webpack
 */
const tsconfig = require('./typescript/tsconfig');

const transformer: TsConfigTransformer = (config: TypescriptConfigMutator) => {
  config
    .mergeTsConfig(tsconfig)
    .setArtifactName('declaration')
    .setDistGlobPatterns([`dist/**/*.d.ts`])
    .setShouldCopyNonSupportedFiles(true);
  return config;
};

export class PlasmaReactMain {
  static slots = [];

  static dependencies = [ReactAspect, EnvsAspect];

  static runtime = MainRuntime;

  static async provider([react, envs]: [ReactMain, EnvsMain]) {
    const templatesReactEnv = envs.compose(react.reactEnv, [
      /**
       * Uncomment to override the config files for TypeScript, Webpack or Jest
       * Your config gets merged with the defaults
       */

      react.useTypescript({ buildConfig: [transformer] }),
      // react.useWebpack({
      //   previewConfig: [previewConfigTransformer],
      //   devServerConfig: [devServerConfigTransformer],
      // }),
      // react.overrideJestConfig(require.resolve('./jest/jest.config')),

      /**
       * override the ESLint default config here then check your files for lint errors
       * @example
       * bit lint
       * bit lint --fix
       */
      // react.useEslint({
      //   transformers: [ sad
      //     (config) => {
      //       config.addExtends(['@plasma/ui.ts-recommended/react']);
      //       // config.setPluginPath('@plasma/ui.eslint-config-ts-recommended/react');
      //       return config;
      //     },
      //   ],
      // }),

      /**
       * override the Prettier default config here the check your formatting
       * @example
       * bit format --check
       * bit format
       */
      react.usePrettier({
        transformers: [
          (config) => {
            config.setKey('tabWidth', 2);
            return config;
          },
        ],
      }),

      /**
       * override dependencies here
       * @example
       * Uncomment types to include version 17.0.3 of the types package
       */
      react.overrideDependencies({
        dependencies: {},
        devDependencies: {},
        peerDependencies: {},
      }),
    ]);
    envs.registerEnv(templatesReactEnv);
    return new PlasmaReactMain();
  }
}

PlasmaReactAspect.addRuntime(PlasmaReactMain);
`

tsconfig file that is being used on the main.runtime file
`{
  "include": ["./**/i18n/*.json", "**/*.ts", "**/*.tsx" ]
}
`

**Comment by KutnerUri** [comment]
a couple of issues:

- you are working "ownerless", ie `"defaultScope": "plasma"` instead of `"defaultScope": "owner.plasma"`. This should be fine, but might have bugs we don't know about.
- `overrideDependencies({ dependencies: {}, devDependencies: {}, peerDependencies: {} })` - I think this could be removing all dependencies from your components.

I don't see anything else.
Maybe there is a (hidden) error during `bit compile` that prevents it from completing and generating dists?

please verify `bit compile` is successful by looking at `node_modules/@plasma/ui.themes.theme-provider`. It should have:
- a `dist` folder with the compiled files.
- a `"main": "dist/index.js"` entry in the package.json

![Screen Shot 2022-07-05 at 22 44 12](https://user-images.githubusercontent.com/5400361/177403826-5b69842a-70cd-48d0-bd98-34fb4c990f4a.png)

**Comment by lhassis** [comment]
@KutnerUri after your tip about the dist folder, I checked the capsules and found out the problem was on my override of tsconfig.

```tsx
const transformer: TsConfigTransformer = (config: TypescriptConfigMutator) => {
  config
    .mergeTsConfig(tsconfig)
    .setArtifactName('declaration')
    .setDistGlobPatterns([`dist/**/*.d.ts`])
    .setShouldCopyNonSupportedFiles(true);
  return config;
};
```

The `setDistGlobPatterns` was forcing to only generate the .d.ts files. I think I copied it carelessly from the https://bit.cloud/teambit/typescript/typescript and didn't realize what it would cause. By removing this, I was able to update the versions.

About the dependency override, I guess it is not clearing it, because the package.json files are created with the correct dependencies.

Thanks a lot for the help!
